### ISSUE TITLE
Could you offer detail of environment?

### ISSUE THREAD

**Issue by LiJunscs** [OpenGVLab/VideoChat-Flash#20]
Thank you for your work. But when I use the code to evaluate the longvideobench, the generation of model is always "!".
I ever encountered the problem when I evaluate nvila in the same benchmark, I change the flash attention version to solve it.
But in your work, I can't solve it. Could you offer a solution?

**Comment by leexinhao** [comment]
What's your transformers version? And which model do you use (UMT-based or InternVideo2-based) ?

**Comment by LiJunscs** [comment]
> What's your transformers version? And which model do you use (UMT-based or InternVideo2-based) ?

![Image](https://github.com/user-attachments/assets/cd9f590c-90b3-4942-adce-6c9bc366d033)

The important packages are above in my environment.
The model I use is the VideoChat-Flash-Qwen2-7B_res448 downloaded from huggingface according to your url.

**Comment by leexinhao** [comment]
Can you inference normally with the demo code in huggingface README?

**Comment by LiJunscs** [comment]
> Can you inference normally with the demo code in huggingface README?
```
import os
import sys
current_dir = os.path.dirname(os.path.abspath(__file__))
project_root = os.path.dirname(current_dir)
if project_root not in sys.path:
    sys.path.append(project_root)
from transformers import AutoModel, AutoTokenizer
from videochat_flash.modeling_videochat_flash import VideoChatFlashQwenForCausalLM

# model setting
model_path = 'model.ckpt'

tokenizer = AutoTokenizer.from_pretrained(model_path, trust_remote_code=True)
model = VideoChatFlashQwenForCausalLM.from_pretrained(model_path, trust_remote_code=True).half().cuda()
image_processor = model.get_vision_tower().image_processor

mm_llm_compress = False # use the global compress or not
if mm_llm_compress:
    model.config.mm_llm_compress = True
    model.config.llm_compress_type = "uniform0_attention"
    model.config.llm_compress_layer_list = [4, 18]
    model.config.llm_image_token_ratio_list = [1, 0.75, 0.25]
else:
    model.config.mm_llm_compress = False

# evaluation setting
max_num_frames = 512
generation_config = dict(
    do_sample=False,
    temperature=0.0,
    max_new_tokens=1024,
    top_p=0.1,
    num_beams=1
)

video_path = "xxx.mp4"

# single-turn conversation
question1 = "Describe this video in detail."
output1, chat_history = model.chat(video_path=video_path, tokenizer=tokenizer, user_prompt=question1, return_history=True, max_num_frames=max_num_frames, generation_config=generation_config)

print(output1)

# multi-turn conversation
question2 = "How many people appear in the video?"
output2, chat_history = model.chat(video_path=video_path, tokenizer=tokenizer, user_prompt=question2, chat_history=chat_history, return_history=True, max_num_frames=max_num_frames, generation_config=generation_config)

print(output2)
```

This is my demo download from huggingface. I modify the import path, and copy the modeling_qwen2_flash.py and modeling_videochat_flash.py from the huggingface cache folder to the same directory. Because my huggingface cache file is for version 4.46.

My flash attention version is flash_attn-2.7.2.post1+cu12torch2.1cxx11abiFALSE-cp310-cp310-linux_x86_64.whl downloaded from flash attention repo.

But the response of demo is alse "!". Like follow:

![Image](https://github.com/user-attachments/assets/27b1b335-a7c2-4cbb-bf99-8e00782458ca)

**Comment by leexinhao** [comment]
Try update transformers==4.40.1.

**Comment by LiJunscs** [comment]
> Try update transformers==4.40.1.

It's still no use. And I think the problem is not the version of transformers.
Here is a screenshot of the process of generate.

![Image](https://github.com/user-attachments/assets/3d8c9173-377e-46cf-897d-0d0941681bc6)

In the image, we can see that the forward is finished normally, but the output logits is nan. It maybe version of flash attention or other related to the calculation.

My gpu is A100 80G, flash attention is flash_attn-2.7.2.post1+cu12torch2.1cxx11abiFALSE-cp310-cp310-linux_x86_64.whl.

**Comment by leexinhao** [comment]
try to close flash attn? https://huggingface.co/OpenGVLab/VideoChat-Flash-Qwen2_5-2B_res448/blob/aac07d6308835645455d1c228b1fbeeafa105151/vision_tower_builder.py#L69
modify `flash` to `origin`

**Comment by LiJunscs** [comment]
> try to close flash attn? https://huggingface.co/OpenGVLab/VideoChat-Flash-Qwen2_5-2B_res448/blob/aac07d6308835645455d1c228b1fbeeafa105151/vision_tower_builder.py#L69 modify `flash_v2` to `origin`

Ohhh, I solve it. The demo code and your code download from huggingface use the torch.float16 default。 At the last layer, some attention scores exceed the representable range of torch.float16. I change the type of model and frames to torch.bfloat16, it can work normally.

Like follow:

- demo.py:
```python
model = VideoChatFlashQwenForCausalLM.from_pretrained(model_path, trust_remote_code=True).to(dtype=torch.bfloat16, device="cuda:0")
model = model.eval()
```

- modeling_videochat_flash.py
```python
frames, time_msg  = load_video(video_path, max_num_frames=max_num_frames, media_dict=media_dict)

image_sizes = [frames[0].shape[:2]]

frames = [self.get_vision_tower().image_processor.preprocess(frames, return_tensors="pt")["pixel_values"].to(dtype=torch.bfloat16).cuda()]
```

Good luck !!!

**Comment by LiJunscs** [comment]
> > try to close flash attn? https://huggingface.co/OpenGVLab/VideoChat-Flash-Qwen2_5-2B_res448/blob/aac07d6308835645455d1c228b1fbeeafa105151/vision_tower_builder.py#L69 modify `flash_v2` to `origin`
> 
> Ohhh, I solve it. The demo code and your code download from huggingface use the torch.float16 default。 At the last layer, some attention scores exceed the representable range of torch.float16. I change the type of model and frames to torch.bfloat16, it can work normally.
> 
> Like follow:
> 
> * demo.py:
> 
> model = VideoChatFlashQwenForCausalLM.from_pretrained(model_path, trust_remote_code=True).to(dtype=torch.bfloat16, device="cuda:0")
> model = model.eval()
> * modeling_videochat_flash.py
> 
> frames, time_msg  = load_video(video_path, max_num_frames=max_num_frames, media_dict=media_dict)
> 
> image_sizes = [frames[0].shape[:2]]
> 
> frames = [self.get_vision_tower().image_processor.preprocess(frames, return_tensors="pt")["pixel_values"].to(dtype=torch.bfloat16).cuda()]
> Good luck !!!
